### ISSUE TITLE
Defining type: 'compute' throws when setting null/undefined ~3

### ISSUE THREAD

**Issue by dylanrtt** [canjs/canjs#2372]
Defining a map with `type: 'compute'` throws when setting a value such as `null` or `undefined` here: [map/define/define.js#L239](https://github.com/canjs/canjs/blob/bbbb15f6a8def6f41af6b5788ab7dc80fc43fa95/map/define/define.js#L239). Probable fix is to guard with `newValue &&`.

Example:

```js
var VM = can.Map.extend({ define: {
    foo: { type: 'compute' }
}});

// throws
var vm = new VM({ foo: null });
```

http://jsbin.com/focinurivu/edit?js,console

Other non-compute values work and I think this should be supported.

**Comment by justinbmeyer** [comment]
Fix this first: https://github.com/canjs/canjs/issues/2316

Then make sure this works in:

- [ ] - can-define/map
- [ ] - can-map-define

**Comment by bmomberger-bitovi** [comment]
can-define/map doesn't throw an error after the fix for #2316, but can-map-define still does.

**Comment by garrettreed** [comment]
can-map-define is fixed in this PR: https://github.com/canjs/can-map-define/pull/12

can-define/map is no longer marked as fixed because the compute getter/setter is never called. There are no compute tests, so everything was passing. The same fix used for can-map-define can be implemented when computes are working.
